### PR TITLE
At-A-Glance: Only show the Scan card for multisites if they're connected to VaultPress

### LINKED ISSUE #17220: Plans: adjust wording for all upgrades when site is part of a Multisite network
### Steps to reproduce the issue 

1. Start from a site that's the main site of a Multisite network.
2. Connect that site to WordPress.com.
3. Go to Jetpack > Dashboard in wp-admin
4. Click on the prompts to upgrade
5. You'll be able to purchase a Jetpack Complete Plan
6. Upon purchase, and when back to wp-admin, you'll still see prompts to upgrade in the AAG dashboard as well as in Jetpack > Settings.

The call to `jetpack/v4/rewind` returns the following:

```
{"code":"success","message":"Backup &amp; Scan data correctly received.","data":"{\"reason\":\"multisite_not_supported\",\"state\":\"unavailable\",\"last_updated\":\"2020-09-21T17:03:10.651+00:00\",\"has_cloud\":true}"}
```

- [ ] It seems like we should take such Rewind states into account better.
- [ ] We may want to update the purchase prompts and flows so we do not even get into that state.

**Screenshots**

image (10)

Team51 project, backups and scan on settings page also entirely blank. 

CleanShot 2020-09-21 at 09 21 14@2x

### PR DESCRIPTION
**Pull request by mattgawarecki**
Fixes #17220.

#### Changes proposed in this Pull Request:

* Significantly refactor the `DashScan` component to make the VaultPress/Rewind switching logic more straightforward
* Add copy and handling to the Scan card for states where VaultPress is active but not registered/connected
* Use Rewind status to determine whether the site uses VaultPress or Rewind, instead of checking VaultPress data directly (doesn't appear to be as reliable?)
* Hide the At-A-Glance Scan card entirely for multi-sites that don't use VaultPress, since Rewind doesn't currently support multi-site networks

#### Jetpack product discussion

N/A

#### Does this pull request change what data or activity we track or use?

Nope 👍 

##### All scenarios

Monitor the **Network** panel in your browser's developer console. Verify that outgoing requests eventually stop once the page is fully loaded; or, to rephrase, ensure that network calls to get VaultPress/Rewind status don't infinitely repeat.

##### Scenario 1: Multi-site with Jetpack Scan via Rewind

**NOTE:** These instructions should be true whether or not the VaultPress plugin is currently installed or active.

* Open wp-admin for the primary site in a multi-site network. This site should have a purchase that would normally give access to Jetpack Scan (i.e., Jetpack Premium, Jetpack Security, etc.), and it should be set to use Rewind instead of VaultPress.
* Verify that on the **Jetpack > Dashboard** page, the Scan card is not visible.

Screen Shot 2020-09-29 at 10 41 32

##### Scenario 2: Multi-site with Jetpack Scan via VaultPress

* Open wp-admin for the primary site in a multi-site network. This site should have a purchase that would normally give access to Jetpack Scan (i.e., Jetpack Premium, Jetpack Security, etc.), and it should be set to use VaultPress.
* Starting **without** the VaultPress plugin installed, go through and verify the following behaviors on the **Jetpack > Dashboard** page:
  * Plugin not installed: Scan card prompts user to install/activate the plugin
  * Plugin installed but not active: Scan card prompts user to install/activate the plugin (verify link goes to VaultPress install page)
  * Plugin installed and active, site not registered with VaultPress: Scan card prompts the user to register the site with VaultPress (verify links to go VaultPress admin page and VaultPress help page)
  * Plugin active and VaultPress registered, has threats: Scan card shows that threats were found
  * Plugin active and VaultPress registered, no threats: Scan card shows no threats were found

Screen Shot 2020-09-29 at 10 37 40
Screen Shot 2020-09-29 at 10 35 17
Screen Shot 2020-09-29 at 13 35 19
Screen Shot 2020-09-29 at 10 36 21

##### Scenario 3: Single site with Jetpack Scan via Rewind

**NOTE:** These instructions should be true whether or not the VaultPress plugin is currently installed or active.

* Open wp-admin for a site that's not part of any multi-site network. This site should be set to use Rewind instead of VaultPress.
* Verify the following behaviors on the **Jetpack > Dashboard** page:
  * Jetpack Scan not purchased or included via another purchase: Scan card shows a prompt to purchase Scan
  * Jetpack Scan purchased but credentials not entered: Scan card asks the user to enter their credentials to complete the process
  * Jetpack Scan purchased and credentials entered, no threats: Scan card shows "no threats found"
  * Jetpack Scan purchased and credentials entered, threats found: Scan card shows how many threats were found

Screen Shot 2020-09-29 at 10 29 27
Screen Shot 2020-09-29 at 10 22 56
Screen Shot 2020-09-29 at 10 24 15
Screen Shot 2020-09-29 at 10 27 59

##### Scenario 4: Single site with Jetpack Scan via VaultPress

* Open wp-admin for a site that's not part of any multi-site network. This site should have a purchase that would normally give access to Jetpack Scan (i.e., Jetpack Premium, Jetpack Security, etc.), and it should be set to use VaultPress.
* Follow instructions for **Scenario 2**.

#### Proposed changelog entry for your changes:

* On the Dashboard, hide the Scan card for non-VaultPress multi-site networks (as they're not currently supported), and add copy to handle when VaultPress is not connected/registered.

### CODE REVIEW

**Comment by mattgawarecki** [comment]
@Automattic/jetpack-design Could y'all take a look at the screenshots I've attached to this PR to make sure the copy and everything looks good? Specifically, these two things:

1. I've added copy to handle VaultPress customers who are unregistered for whatever reason (all the rest should be the same as they were).
2. I'm hiding the Scan card for multi-sites that use Rewind, instead of showing a notice that Scan isn't compatible with multi-sites. I'm fine to do it either way, just let me know which you'd prefer.

**Comment by mattgawarecki** [comment]
Looks like codeclimate isn't a fan of me using early returns as a readability tool. Should I re-write to follow a single-return pattern, or is this safe to ignore?

**Comment by carolynannewells** [comment]
Hello! Carolyn from editorial here! I have done a copy review and have some suggestions. If you have anything else you would like me to look at please let me know. 👍 > For automated, comprehensive scanning of security threats, please install and activate VaultPress. 

**Scenario 2**

> For automated, comprehensive scanning of security threats, please install and activate VaultPress. 

No security plugin installed. Please _install VaultPress_ to automatically scan for security threats. 

>VaultPress is having trouble connecting. Make sure you’ve entered your registration key, or contact support if you still need help. 

VaultPress is having a problem scanning. Please make sure your _registration key is entered_. If you require further assistance please _contact support_.

>Security threats found. Please fix these as soon as possible. 

Security threats found. Please _secure your network_ immediately.

**Scenario 3**

>Automated scanning and one-click fixes keep your site ahead of security threats. 

Purchase Jetpack Scan to protect your site from security threats with automated scanning. 

>You need to enter your server’s credentials to finish the setup. 

Please finish your setup by _entering your server’s credentials_.

>We are making sure your site stays free of security threats. You will be notified if we find one. 

No security threats found. Your site will continue to be monitored for future threats. 

>Security threat found. Please fix it as soon as possible. 

Security threat found. Please _secure your site_ immediately.

**Comment by jeherve** [comment]
> Looks like codeclimate isn't a fan of me using early returns as a readability tool. Should I re-write to follow a single-return pattern, or is this safe to ignore?

It's fine to ignore I think.

File: _inc/client/at-a-glance/scan.jsx [hunk 1]
```diff
@@ -106,51 +108,75 @@ class DashScan extends Component {
...
+			return renderCard( {
+				className: 'jp-dash-item__is-inactive',
+				status: 'not-registered',
+				content: jetpackCreateInterpolateElement(
+					__(
+						"VaultPress is having trouble connecting. Make sure you've <keyLink>entered your registration key</keyLink>, or <supportLink>contact support</supportLink> if you still need help.",
+						'jetpack'
+					),
+					{
+						keyLink: <a href="admin.php?page=vaultpress" />,
```
Comment (jeherve) [inline _inc/client/at-a-glance/scan.jsx:124]: Maybe we could use `siteAdminUrl` from the props here, to ensure we always build the right URL?

**Review by jeherve — CHANGES_REQUESTED** [review]

**Comment by keoshi** [comment]
Whoa, this is really comprehensive. Thanks for all the consolidation on the product but also for the detailed instructions!

My initial thought was that hiding the Scan card on multi-site networks seemed like a bad idea, but the more I think about the more sense it makes. No one would like to be reminded _ad aeternum_ that you can't own a product because of your setup that can't simply be changed anyhow. Hiding the Scan card in these instances then seems to set the right expectations for the onset. 👍

> Scenario 2: Multi-site with Jetpack Scan via VaultPress

I have a question about this scenario, but first I'll list my assumptions to make this extra clear:

- VaultPress is compatible with sites on multi-site networks.
- Jetpack Scan (via Rewind) is not.
- VaultPress is available as a plugin, but you can't purchase new VP subscriptions.
- All VaultPress' flows direct you to purchasing a Jetpack Backup plan.

If the above is correct, it brings up this point and follow-up questions:

> - Plugin not installed: Scan card prompts user to install/activate the plugin

- Is this scenario targeted at existing VaultPress users that already have a subscription or key?
- Could we be creating a scenario where non-existing customers will install VP, get redirected to purchase Jetpack Scan instead, and end up on wp-admin without a card for the product that they just purchased (and isn't compatible with their site)?

**Comment by keoshi** [comment]
Thanks for the thorough review @carolynannewells ! A couple of notes on the copy suggestions:

> No security plugin installed. Please install VaultPress to automatically scan for security threats.

The “No security plugin installed” part could be ambiguous. Are we communicating any security plugin will do it? The clarification afterwards looks good, but I'm wondering if the two portions could be more connected.

> Security threats found. Please secure your network immediately.

Saying “secure your network” feels very generic and open to interpretation. Is it a problem on my server? My network of sites? This is a very specific issue that we can track back to threats, so being accurate in our description could go a long way towards making them feel confident about the whole process.

Everything else looks wonderful, particularly all the improvements to scenario 3!

**Comment by carolynannewells** [comment]
Thank you for the feedback @keoshi. Would this phrasing work better for you? Let me know if this still isn't conveying the correct meaning. 

VaultPress is not installed, _please install_ to enable automatic scanning for security for threats. 

Security threats found. _Click here_ to fix them immediately.

**Comment by keoshi** [comment]
That's stellar, @carolynannewells ! Thank you again. 🙌

**Comment by mattgawarecki** [comment]
@keoshi

> Thanks for all the consolidation on the product but also for the detailed instructions!

Thanks for testing! I know it's a lot 😅 , so I appreciate you taking the time.

> > Scenario 2: Multi-site with Jetpack Scan via VaultPress
> 
> I have a question about this scenario, but first I'll list my assumptions to make this extra clear:
> 
>     * VaultPress is compatible with sites on multi-site networks.
> 
>     * Jetpack Scan (via Rewind) is not.
> 
>     * VaultPress is available as a plugin, but you can't purchase new VP subscriptions.
> 
>     * All VaultPress' flows direct you to purchasing a Jetpack Backup plan.

These are all correct, to my understanding.

> If the above is correct, it brings up this point and follow-up questions:
> 
> > * Plugin not installed: Scan card prompts user to install/activate the plugin
> 
>     * Is this scenario targeted at existing VaultPress users that already have a subscription or an unused key?

Yes, that's exactly what I'm targeting here. Specifically what we're keying on is the fact that the site's Rewind status amounts to "Unavailable: this site is currently assigned to use VaultPress," which would only happen if they already have a paid-up subscription or if they've been manually switched away from Rewind for some reason.

>     * Could we be creating a scenario where non-existing customers will install VP, get redirected to purchase Jetpack Scan instead, and end up on wp-admin without a card for the product that they just purchased (and isn't compatible with their site)?

I made sure not to rely on the status of the VaultPress plugin as any indication of which service the site is using. Since you should only ever see VaultPress messages if you're already a VaultPress customer, I don't see this happening. :-)

**Comment by mattgawarecki** [comment]
@keoshi @carolynannewells 

With your suggested copy revisions, I also separated the two cases for "inactive plugin" vs "not installed plugin." Do these look okay to you?

Screen Shot 2020-10-01 at 10 27 46

Screen Shot 2020-10-01 at 10 28 40

**Comment by mattgawarecki** [comment]
Ready to test again, everybody 👍

**Comment by keoshi** [comment]
> I made sure not to rely on the status of the VaultPress plugin as any indication of which service the site is using. Since you should only ever see VaultPress messages if you're already a VaultPress customer, I don't see this happening. :-)

That's very sensible though I'm not sure how flows are equally as. 😅

Meaning you can be an existing VP customer, not have any spare subscription keys, and wanting to purchase one. The problem here is that you cannot purchase a VP subscription anymore, and all the flows on vaultpress.com lead to the new Jetpack plans. Hence my question if a customer could end up effectively with an hidden card for a product they just purchased.

I **_don't_** know that this is 100% true, so pinging @singerb who is really insightful and helpful breaking down these scenarios. Thanks in advance, Benedict!

> With your suggested copy revisions, I also separated the two cases for "inactive plugin" vs "not installed plugin."

Super nice! The more clarity, the better. 🙌

File: _inc/client/at-a-glance/scan.jsx [hunk 2]
```diff
@@ -106,55 +110,99 @@ class DashScan extends Component {
...
+		// VaultPress is installed, just not activated
+		if ( this.props.isVaultPressInstalled ) {
+			return renderCard( {
+				className: 'jp-dash-item__is-inactive',
+				status: 'pro-inactive',
+				content: [
+					<p className="jp-dash-item__description" key="inactive-scanning">
+						{ jetpackCreateInterpolateElement(
+							__(
+								'VaultPress is not active, <a>please activate</a> to enable automatic scanning for security for threats.',
```
Comment (jeherve) [inline _inc/client/at-a-glance/scan.jsx:184]: We do have an endpoint you could use here to help folks activate the plugin right from there. Here is an example with Akismet
https://github.com/Automattic/jetpack/blob/e20534fc3d97faa58db9252a030f2f5585b306be/_inc/client/at-a-glance/akismet.jsx#L62-L63

**Comment by singerb** [comment]
If we have a backend site that's assigned to use VaultPress, but the plugin is not installed on that site, I think it's fine to show the card and prompt. And trusting the Scan status to indicate that Scan is unavailable because the site is using VP is the way to go there. How many VP keys the customer has doesn't affect that.

File: _inc/client/at-a-glance/scan.jsx (hunk 2 above)
Comment (mattgawarecki) [reply _inc/client/at-a-glance/scan.jsx:214]: Oh, that's slick! Added the same functionality to this PR. 👍

**Review by jeherve — APPROVED** [review]
This tests well for me. 🚢

**Pull request merged**